mobile/mobile-core/android-process-service.ts

```diff
@@ -0,0 +1,189 @@
+			// Num       		 RefCount Protocol Flags    Type St Inode  Path
+			// 0000000000000000: 00000002 00000000 00010000 0001 01 189004 @webview_devtools_remote_25512
+			// The Path column is the abstract port.
+			let abstractPortRegex = /\w+:\W+(\w+[ \t]+){1,6}/;
+
+			// Need to search by processId to check for old Android webviews (@webview_devtools_remote_<processId>).
+			let androidWebViewPortInformation = adb.executeShellCommand(["cat", "proc/net/unix", "|grep", "-a", "--text", processId]).wait();
+			// Search for appIdentifier (@<appIdentifier>_devtools_remote).
+			let chromeWebViewPortInformation = adb.executeShellCommand(["cat", "proc/net/unix", "|grep", "-a", "--text", appIdentifier + "_devtools_remote"]).wait();
+			// Search for appIdentifier (@<appIdentifier-debug>).
+			let nativeScriptWebViewPortInformation = adb.executeShellCommand(["cat", "proc/net/unix", "|grep", "-a", "--text", `${appIdentifier}-debug`]).wait();
+
+			// Need to check for
+			let abstractPortInformation: string = androidWebViewPortInformation || chromeWebViewPortInformation || nativeScriptWebViewPortInformation;
+
+			if (!abstractPortInformation.match(abstractPortRegex)) {
+				return null;
+			}
+
+			// Need to take the third element from the array because after the split the array will conatin ['', '<Match from the regex group>', '<AbstractPort>'].
+			// Abstract port will be in format @<AbstractPort> and need to remove the "@" character.
+			return abstractPortInformation.split(abstractPortRegex)[2].trim().substr(1);
+		}).future<string>()();
+	}
+
+	private getProcessId(adb: Mobile.IDeviceAndroidDebugBridge, appIdentifier: string): IFuture<string> {
+		return (() => {
+			// Process information will look like this (without the columns names):
+			// USER     PID   PPID  VSIZE   RSS   WCHAN    PC         NAME
+			// u0_a63   25512 1334  1519560 96040 ffffffff f76a8f75 S com.telerik.appbuildertabstest
+			let processIdRegex = /^\w*\W*([0-9]+)/;
+			let processIdInformation: string = adb.executeShellCommand(["ps", "|grep", "-a", "--text", appIdentifier]).wait();
+
+			let matches = processIdRegex.exec(processIdInformation);
+
+			if (!matches || !matches[0]) {
+				return null;
+			}
+
+			return matches[1];
+		}).future<string>()();
+	}
+
+	private getAvailableAndroidPortsInformation(adb: Mobile.IDeviceAndroidDebugBridge, appIdentifier: string): IFuture<Mobile.IAndroidPortInformation[]> {
+		return (() => {
+			// Get ports information.
+			// Result will look like this:
+			// sl local_address rem_address   st tx_queue rx_queue tr tm->when retrnsmt  uid  timeout inode
+			// 0: 0100007F:1C23 00000000:0000 0A 00000000:00000000 00:00000000 00000000  1001       0 2111 1 e22cc000 300 0 0 2 -1
+			let tcpPorts: string[] = adb.executeShellCommand(["cat", "proc/net/tcp"]).wait().split(EOL);
+			let tcp6Ports: string[] = adb.executeShellCommand(["cat", "proc/net/tcp6"]).wait().split(EOL);
+
+			let allPorts: string[] = tcpPorts.concat(tcp6Ports);
```

Comment:
is it possible `tcpPorts` to be undefined or null. In case yes, this should be:
```TypeScript
let allPorts = (tcpPorts || []).concat(tcp6ports || []);
```